pkg/datapath/maps/map.go

```diff
@@ -95,9 +95,14 @@ func newMapSweeper(g endpointManager) *mapSweeper {
 // deleteMapIfStale uses the endpointManager implementation to determine for
 // the given path whether it should be deleted, and if so deletes the path.
 func (ms *mapSweeper) deleteMapIfStale(path string, filename string, endpointID string) {
-	if tmp, err := strconv.ParseUint(endpointID, 0, 16); err == nil {
+	if tmp, err := strconv.ParseUint(endpointID, 10, 16); err == nil {
 		epID := uint16(tmp)
-		if !ms.endpointExists(epID) {
+		if ms.endpointExists(epID) {
+			prefix := strings.TrimSuffix(filename, endpointID)
+			if filename != bpf.LocalMapName(prefix, uint16(tmp)) {
```

Comment:
why not use epID instead of doing `uint16(tmp)` again?

Reply:
;) Probably because this code was shuffled around a couple of times by the time the dependent PRs were merged. Will fix.